`suggest_call_as_method` in HIR type check hints incorrect code when dealing with macros

### Code

```Rust
fn main() {
    let hello = len(vec![]);
}
```


### Current output

```Shell
error[E0425]: cannot find function `len` in this scope
 --> src/main.rs:2:17
  |
2 |     let hello = len(vec![]);
  |                 ^^^ not found in this scope
  |
help: use the `.` operator to call the method `len` on `&Vec<_>`
 --> /home/milomoisson/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:44:36

44-         $crate::__rust_force_expr!($crate::vec::Vec::new())
44+         $crate::__rust_force_expr!(.len())
  |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `rustc-wrong-hint-repro` (bin "rustc-wrong-hint-repro") due to previous error
```


### Desired output

```Shell
error[E0425]: cannot find function `len` in this scope
 --> src/main.rs:2:17
  |
2 |     let hello = len(vec![]);
  |                 ^^^ not found in this scope
  |
help: use the `.` operator to call the method `len` on `&Vec<_>`
 --> /home/milomoisson/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:44:36

44-         len(vec![])
44+         vec![].len()
  |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `rustc-wrong-hint-repro` (bin "rustc-wrong-hint-repro") due to previous error
```


### Rationale and extra context

_No response_

### Other cases

_No response_

### Anything else?

rustc help hint with color:
![image](https://github.com/rust-lang/rust/assets/43814157/f2ac6cab-1973-40e1-b8b5-0b327b5bca2e)

--- a/compiler/rustc_hir_typeck/src/callee.rs
+++ b/compiler/rustc_hir_typeck/src/callee.rs
@@ -531,8 +531,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 return;
             }
 
-            let up_to_rcvr_span = segment.ident.span.until(callee_expr.span);
-            let rest_span = callee_expr.span.shrink_to_hi().to(call_expr.span.shrink_to_hi());
+            let Some(callee_expr_span) = callee_expr.span.find_ancestor_inside(call_expr.span)
+            else {
+                return;
+            };
+            let up_to_rcvr_span = segment.ident.span.until(callee_expr_span);
+            let rest_span = callee_expr_span.shrink_to_hi().to(call_expr.span.shrink_to_hi());
             let rest_snippet = if let Some(first) = rest.first() {
                 self.tcx
                     .sess

--- a/tests/ui/methods/suggest-method-on-call-with-macro-rcvr.rs
+++ b/tests/ui/methods/suggest-method-on-call-with-macro-rcvr.rs
@@ -1,0 +1,6 @@
+// issue: 114131
+
+fn main() {
+    let hello = len(vec![]);
+    //~^ ERROR cannot find function `len` in this scope
+}

--- a/tests/ui/methods/suggest-method-on-call-with-macro-rcvr.stderr
+++ b/tests/ui/methods/suggest-method-on-call-with-macro-rcvr.stderr
@@ -1,0 +1,15 @@
+error[E0425]: cannot find function `len` in this scope
+  --> $DIR/suggest-method-on-call-with-macro-rcvr.rs:4:17
+   |
+LL |     let hello = len(vec![]);
+   |                 ^^^ not found in this scope
+   |
+help: use the `.` operator to call the method `len` on `&Vec<_>`
+   |
+LL -     let hello = len(vec![]);
+LL +     let hello = vec![].len();
+   |
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0425`.

